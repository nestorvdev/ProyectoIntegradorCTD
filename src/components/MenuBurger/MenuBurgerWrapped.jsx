import styled from "styled-components"

export const MenuBurgerWrapper = styled.nav`
  align-items: center;
  background: white;
  display: flex;
  flex-direction: column;
  height: 100vh;
  justify-content: flex-start;
  position: fixed;
  right: ${props => (props.show ? "0" : "-100%")};
  top: 0px;    
  transition: right 0.3s linear;
  width: 73vw;
  z-index:150;
  @media only screen and (min-width: 415px) {
    display:none;
  }

  .mobileHeader{
    background-color: var(--primary-color);
    display: flex;
    flex-direction: column;
    height: 23.5vh;
    justify-content: space-between;
    width:73vw;
  }
  
  .close {
    color: white;
    cursor:pointer;
    font-size: 24px;
    font-weight: var(--font-bold);
    padding:5px 10px;
  }

  .close a{
    text-decoration: none;
  }

  .avatar{
    align-items: flex-end;
    display: flex;
    flex-direction: column;
    justify-self: flex-end;
    padding: 5px 12px;
  }

  .logoMobile{
    align-items: center;
    color: var(--tertiary-color);
    display: flex;
    flex-wrap: wrap;
  }

  .logoMobile p {
    background-color: var(--secondary-color);
    border-radius: 50%;
    color: white;
    font-size: var(--heading3);
    font-weight: var(--font-bold);
    padding: 5px;
  }

  .text h3, .mobileBody p {
    font-size: var(--heading3);
    font-weight: var(--font-bold);    
  }
  
  .text h3.great {
    color:var(--quaternary-color);
    opacity: 0.5;
    text-align: right;
  }
  
  .text h3.name {
    color: var(--secondary-color);
  }
  
  .textMenu{
    color: white;
    font-size: var(--heading2);
    font-weight: var(--font-bold);
    padding:5px 12px;
    text-align: right;
  }

  .mobileBody{
    align-items: flex-end;
    display: flex;
    flex-direction: column;
    height:62.5vh;
    justify-content: flex-start;
    padding: 31px 12px;
    width: 73vw;
  }

  .botones{
    align-items: flex-end;
    display: flex;
    flex-direction: column;
    height: 10vh;
    justify-content: space-between;
  }

  .botones img{
    width:67vw
  }

  .botones p{
    color: var(--secondary-color);
    padding: 1px 5px;
  }

  .botones p:hover{
    background-color: var(--primary-color);
    border-radius:5px;
    color: white;
  }

  .mobileFooter{
    align-items: flex-end;
    display: flex;
    flex-direction: column;
    height: 14vh;
    justify-content: space-evenly;
    padding-right: 20px;
    width:73vw;
  }

  .logOut{
    align-items: flex-end;
    display: flex;
    flex-direction: column;
    height: 2.5vh;
  }

  .logOut div{
    font-size: 12px;
    font-weight: var(--font-medium);
  }

  .logOut div a{
    color: var(--primary-color);
    text-decoration: none;
  }
  
  .iconsMobile{
    align-items: flex-end;
    display: flex;
    height: 6vh;
    justify-content: space-between;
    width: 33vw;
  }

  .iconSocial{
    opacity:0.6;
  }

  .hide{
    display: none;
    text-align:right;
  }

  .subir{
    justify-content:flex-start;
  }

  .menuFavoritos p{
    color: var(--secondary-color);
    text-align: right;
  }

`