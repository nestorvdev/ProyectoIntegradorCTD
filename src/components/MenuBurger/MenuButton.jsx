import React from "react"
import styled from "styled-components"

const MenuButtonWrapper = styled.button`
    background-color:rgb(255,255,255);
    border: none;
    box-shadow: 0px 0px 0px rgb(50, 50, 50);
    cursor:pointer;
    margin-top: 1rem;
    width: clamp(33px,5.2vw,161px);
    
    @media only screen and (min-width: 415px) {
        display: none;
    }
`

export default function MenuButton({ show, handleShow }) {
    return !show ? (
        <MenuButtonWrapper onClick={handleShow}>
            <svg viewBox="0 0 100 80" width="30" height="30" >
                <rect width="100" height="10" />
                <rect y="20" width="100" height="10" />
                <rect y="40" width="100" height="10" />
            </svg>
        </MenuButtonWrapper>
    ) : null
}



