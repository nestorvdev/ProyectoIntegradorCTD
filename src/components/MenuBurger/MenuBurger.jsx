import React, { useState } from "react";
import { Link } from "react-router-dom";
import logoFb from "./img/IconFb.png";
import logoLinkedIn from "./img/IconLinkedIn.png";
import logoTwitter from "./img/IconTw.png";
import logoIg from "./img/IconIg.png";
import line from "./img/Line.png";
import { MenuBurgerWrapper } from "./MenuBurgerWrapped.jsx";
import Share from '../../components/Product/Share';
import iconSocial from "./img/icon-share.svg"

export default function MenuMobile({ admin, handleFavouriteClick, show, handleHide, isLogged, iniciales, activeLogin, activeCreate, handleLogOut }) {

    const hideObject = (isLogged) ? "hide" : null; //Para esconder los botones cuando esta logueado
    const showObject = (isLogged) ? null : "hide"; //Para que se vea el avatar cuando se loguea

    const [shareIsOpenFooter, setShareIsOpenFooter] = useState(false);
    const openShareModal = (() => { setShareIsOpenFooter(true) });
    let placeShareCall = "footer"

    return (
        <MenuBurgerWrapper show={show} className="containerMobile">
            <div className="mobileHeader">
                <div className="close" onClick={handleHide}>X</div>
                <div className={`avatar ${showObject}`}>
                    <div className="logoMobile">
                        <p>{iniciales}</p>
                    </div>
                    <div className="text">
                        <h3 className="great">Hola,</h3>
                        <h3 className="name">{sessionStorage.getItem("name")} {sessionStorage.getItem("surname")}</h3>
                    </div>
                </div>
                <p className={`textMenu ${hideObject}`}>MENÚ</p>
            </div>
            <div className="mobileBody">
                <div onClick={handleHide} className={`menuFavoritos ${!isLogged ? "hide" : null} `}>
                    <Link to="/">
                        <p onClick={handleFavouriteClick}>Favoritos</p>
                    </Link>
                    <img src={line} alt="" />
                    <Link to="/mybookings">
                        <p>Mis reservas</p>
                    </Link>
                    {admin &&
                        <div>
                            <img src={line} alt="" />
                            <Link to="/product/create">
                                <p>Crear Producto</p>
                            </Link>
                            <img src={line} alt="" />
                            <Link to="/product/update">
                                <p>Modificar Producto</p>
                            </Link>
                        </div>
                    }
                </div>
                <div className={`botones ${hideObject} ${(!activeLogin && activeCreate) ? "subir" : null}`}>

                    <Link to="/create" id="createC" onClick={handleHide}><p className={(!activeLogin && activeCreate) ? "hide" : null} >Crear cuenta</p></Link>
                    <img className={`${activeCreate || activeLogin ? "hide" : null}`} src={line} alt="" />
                    <Link to="/login" onClick={handleHide}><p className={(!activeCreate && activeLogin) ? "hide" : null} >Iniciar sesión</p></Link>
                </div>
            </div>
            <div className="mobileFooter">
                <div className={`logOut ${showObject}`}>
                    <div>¿Deseas <a href="/" onClick={handleLogOut}>cerrar sesión?</a></div>
                    <img className="iconsImgMobile" src={line} alt="" />
                </div>
                <div className="iconsMobile">

                    <a href="https://www.facebook.com/profile.php?id=100074545225635" target="_blank" rel="noreferrer" ><img src={logoFb} alt="" /></a>
                    <a href="https://www.linkedin.com/in/world-guest-2b195b225/" target="_blank" rel="noreferrer" ><img src={logoLinkedIn} alt="" /></a>
                    <a href="https://twitter.com/WorldGuest2" target="_blank" rel="noreferrer" > <img src={logoTwitter} alt="" /></a>
                    <a href="https://www.instagram.com/world_guest_booking/" target="_blank" rel="noreferrer"> <img src={logoIg} alt="" /></a>
                    <img src={iconSocial} alt="iconSocial" onClick={openShareModal} className="iconSocial" />
                    <Share shareIsOpen={shareIsOpenFooter} placeShareCall={placeShareCall} setShareIsOpen={setShareIsOpenFooter} />
                </div>
            </div>
        </MenuBurgerWrapper>
    )
}

