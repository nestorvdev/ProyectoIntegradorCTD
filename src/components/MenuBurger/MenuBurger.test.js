import React from "react";
import { shallow } from "enzyme";
import "@testing-library/jest-dom";
import MenuBurger from "./MenuBurger"
import MenuButton from "./MenuButton";
import { MenuBurgerWrapper } from "./MenuBurgerWrapped";

import Enzyme from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { createSerializer } from 'enzyme-to-json';

Enzyme.configure({ adapter: new Adapter() });
expect.addSnapshotSerializer(createSerializer({ mode: 'deep' }));

describe("Probando el componente <MenuBurger/>", () => {
    let wrapperMenu; 
    let wrapperBoton;
    let props={
        show:false,
        handleShow: jest.fn()
    }
    beforeEach(() => {
        wrapperMenu = shallow(<MenuBurger />)
        wrapperBoton = shallow(<MenuButton {...props}/>)
    });

    it("Deberia mostrar <MenuBurger/> correctamente", () => {
        expect(wrapperMenu).toMatchSnapshot();
    });

    it("Deberia retornar una cruz para cerrar", () => {
        let node = wrapperMenu.find("div.close")
        expect(node.length).toEqual(1)
    });

    it("Deberia retornar 2 etiquetas h3", () => {
        let node = wrapperMenu.find("h3")
        expect(node.length).toEqual(2)
    });

    it("Deberia visualizarse el menu al hacer click en el boton", ()=>{
        wrapperBoton.find("svg").simulate("click", props.handleShow)
        expect(wrapperMenu.props("show")).toBeTruthy()
    })
});

describe("Probando el componente <MenuButton/>", ()=>{
    let wrapperBoton; 
    beforeEach(() => {
        wrapperBoton = shallow(<MenuButton />)
    });

    it("Deberia mostrar <MenuButton/> correctamente", () => {
        expect(wrapperBoton).toMatchSnapshot();
    });

})

describe("Probando el componente <MenuBurgerWrapper", ()=>{
    let menuBurgerWrapper; 
    beforeEach(() => {
        menuBurgerWrapper = shallow(<MenuBurgerWrapper />)
    });

    it("Deberia mostrar <MenuBurgerWrapper/> correctamente", () => {
        expect(menuBurgerWrapper).toMatchSnapshot();
    });

})
