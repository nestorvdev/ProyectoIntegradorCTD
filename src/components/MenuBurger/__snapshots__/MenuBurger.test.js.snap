// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando el componente <MenuBurger/> Deberia mostrar <MenuBurger/> correctamente 1`] = `
<styled.nav
  className="containerMobile"
>
  <div
    className="mobileHeader"
  >
    <div
      className="close"
    >
      X
    </div>
    <div
      className="avatar hide"
    >
      <div
        className="logoMobile"
      >
        <p />
      </div>
      <div
        className="text"
      >
        <h3
          className="great"
        >
          Hola,
        </h3>
        <h3
          className="name"
        >
           
        </h3>
      </div>
    </div>
    <p
      className="textMenu null"
    >
      MENÚ
    </p>
  </div>
  <div
    className="mobileBody"
  >
    <div
      className="menuFavoritos hide "
    >
      <Link
        to="/"
      >
        <p>
          Favoritos
        </p>
      </Link>
      <img
        alt=""
        src="Line.png"
      />
      <Link
        to="/mybookings"
      >
        <p>
          Mis reservas
        </p>
      </Link>
    </div>
    <div
      className="botones null null"
    >
      <Link
        id="createC"
        to="/create"
      >
        <p
          className={null}
        >
          Crear cuenta
        </p>
      </Link>
      <img
        alt=""
        className="null"
        src="Line.png"
      />
      <Link
        to="/login"
      >
        <p
          className={null}
        >
          Iniciar sesión
        </p>
      </Link>
    </div>
  </div>
  <div
    className="mobileFooter"
  >
    <div
      className="logOut hide"
    >
      <div>
        ¿Deseas 
        <a
          href="/"
        >
          cerrar sesión?
        </a>
      </div>
      <img
        alt=""
        className="iconsImgMobile"
        src="Line.png"
      />
    </div>
    <div
      className="iconsMobile"
    >
      <a
        href="https://www.facebook.com/profile.php?id=100074545225635"
        rel="noreferrer"
        target="_blank"
      >
        <img
          alt=""
          src="IconFb.png"
        />
      </a>
      <a
        href="https://www.linkedin.com/in/world-guest-2b195b225/"
        rel="noreferrer"
        target="_blank"
      >
        <img
          alt=""
          src="IconLinkedIn.png"
        />
      </a>
      <a
        href="https://twitter.com/WorldGuest2"
        rel="noreferrer"
        target="_blank"
      >
         
        <img
          alt=""
          src="IconTw.png"
        />
      </a>
      <a
        href="https://www.instagram.com/world_guest_booking/"
        rel="noreferrer"
        target="_blank"
      >
         
        <img
          alt=""
          src="IconIg.png"
        />
      </a>
      <img
        alt="iconSocial"
        className="iconSocial"
        onClick={[Function]}
        src="icon-share.svg"
      />
      <Share
        placeShareCall="footer"
        setShareIsOpen={[Function]}
        shareIsOpen={false}
      />
    </div>
  </div>
</styled.nav>
`;

exports[`Probando el componente <MenuBurgerWrapper Deberia mostrar <MenuBurgerWrapper/> correctamente 1`] = `
<nav
  className="sc-bdvvtL eRUdUI"
/>
`;

exports[`Probando el componente <MenuButton/> Deberia mostrar <MenuButton/> correctamente 1`] = `
<styled.button>
  <svg
    height="30"
    viewBox="0 0 100 80"
    width="30"
  >
    <rect
      height="10"
      width="100"
    />
    <rect
      height="10"
      width="100"
      y="20"
    />
    <rect
      height="10"
      width="100"
      y="40"
    />
  </svg>
</styled.button>
`;
